popup.js

```diff
@@ -1,38 +1,81 @@
-let active = document.getElementById('active');
+const active = document.getElementById('active');
+const activeLabel = document.getElementById('on-switch-label');
+const intervalSlider = document.getElementById('reload-int-input');
+const intervalDisplay = document.getElementById('reload-int-display');
+const settingsContainer = document.getElementById('settings-container');
+const seconds = document.getElementById('sec');
+const minutes = document.getElementById('min');
 
+// Maintain checked input box if this window has active reload
 chrome.tabs.query({ currentWindow: true, active: true }, function (tabs) {
   let tab = tabs[0];
   chrome.storage.local.get(tab.id.toString(), (items) => {
-      if(items[tab.id]){
-          console.log("found!");
-          active.checked = true;
-          
+    // Load previous user input settings
+    if (items[tab.id]) {
+      // Load unit check box
+      if (items[tab.id].unit === 'min') {
+        minutes.checked = true;
+      } else {
+        seconds.checked = true;
       }
+
+      // Load Interval display
+      intervalDisplay.innerHTML = items[tab.id].input;
+      intervalSlider.value = items[tab.id].input;
+    }
+
+    // Check off checkbox if currently reloading
+    if(items[tab.id] && items[tab.id].active){
+        active.checked = true;
+    }
   });
 });
 
+// Save current window tab to Chrome storage
 active.onchange = function(event) {
-  if(event.target.checked){
-    let interval = 10000;
-    
+  // Save reload interval parameter to Chrome storage
+  let numInput = parseInt(intervalSlider.value);
+  // Toggle to minutes if the option was selected
+  let unitInput = minutes.checked ? 'min' : 'sec';
+
+  // Turn ON
+  if (event.target.checked) {
     chrome.tabs.query({ currentWindow: true, active: true }, function (tabs) {
-      console.log(tabs[0]);
       let tab = tabs[0];
-      let tostore = {};
-      tostore[tab.id] = true;
-      chrome.storage.local.set(tostore);
+      let storedObj = {};
+      storedObj[tab.id] = {
+        active: true,
+        input: numInput,
+        unit: unitInput
+      };
+      chrome.storage.local.set(storedObj);
     });
+
+    activeLabel.innerHTML = 'ON';
+    settingsContainer.classList.add('disabled');
```

Comment:
If we are going to disable, we should indeed show the user that it is disabled with some button popup kind of thing else it feels like a bug. The user when tries to click on the unit button or slider when it is disabled, they see the error: "Disable to modify". 
Some kind of dynamic error toast like this (check the snapshot below):
![error_example](https://user-images.githubusercontent.com/11269256/95540305-1e5a8200-0a0e-11eb-8203-0ac60c2fee9f.gif)


Also why is this disabled? Is there some irregularity that you saw if we keep it enabled?

Reply:
Please refer to my comment below for why I disabled it. Do you know what element this little popup is? I've only seen this in validating a form field using the `require` tag on an `input` element.

Reply:
I could also change the colour of the unit switch buttons and slider to grey to give a better visual indication that it's disabled.

Reply:
Okay, let us grey out them.